src/modules/ods/ods.service.ts

```diff
@@ -47,10 +55,51 @@ export class OdsService {
       if (err instanceof NotFoundException) {
         this.logger.warn(err);
         throw err;
-      } else {
-        this.logger.error(err);
-        throw new InternalServerErrorException('Error trying to find a customer');
       }
+
+      this.logger.error(err);
+      throw new InternalServerErrorException('Error trying to find a customer');
+    }
+  }
+
+  /**
+   * Finds a deal in ODS based on provided deal ID
+   * @param {string} dealId The deal ID to search for
+   *
+   * @returns {Promise<GetOdsDealResponse>} The deal response
+   * @throws {InternalServerErrorException} If there is an error trying to find a deal
+   * @throws {NotFoundException} If no matching deal is found
+   */
+  async findDeal(dealId: string): Promise<GetOdsDealResponse> {
+    try {
+      const storedProcedureInput = this.createOdsStoredProcedureInput(ODS_ENTITIES.DEAL, { deal_code: dealId });
+
+      const storedProcedureResult = await this.callOdsStoredProcedure(storedProcedureInput);
+
+      const storedProcedureJson: OdsStoredProcedureOutputBody = JSON.parse(storedProcedureResult);
+
+      if (storedProcedureJson?.status !== 'SUCCESS') {
+        this.logger.error('Error finding a deal from ODS stored procedure, output: %o', storedProcedureResult);
+        throw new InternalServerErrorException('Error trying to find a deal');
+      }
+
+      if (storedProcedureJson?.total_result_count === 0) {
+        throw new NotFoundException('No deal found');
+      }
+
+      return {
```

Comment:
```suggestion
      return {
        dealId,
        name,
        description,
      }
```

Can we please keep the `return` clean by instanting the variables above.